### PR TITLE
fix(tasks): migrate reorder route onto locked-batch primitive — Phase 4 of task board crash-prevention epic

### PR DESCRIPTION
**Pull request by 2witstudios**
## Summary

- Replaces the N-sequential, unordered per-row `UPDATE` transaction loop in `apps/web/src/app/api/pages/[pageId]/tasks/reorder/route.ts` with the shared `computeReorderPlan` + `lockedBatchReorder` primitive from Phase 3 (`@pagespace/lib/services/reorder`, already merged to `master`).
- This is the actual call site that deadlocked production Postgres on 2026-07-18 (~7-minute deadlock storm on `task_items.position`, 17:30-17:37 UTC). The new primitive locks target rows `FOR UPDATE` in ascending-id order and writes all positions as a single batched `UPDATE ... FROM (VALUES ...)` statement, closing the interleaving window that let concurrent reorders deadlock.
- Request/response contract is unchanged: `{ tasks: [{id, position}] }` in, `{success: true}` out, same broadcast + audit log + activity log side effects.

## `scopeWhere` decision

`task_items` has no direct FK to its task list — list membership is derived the same way the sibling `GET` route already does it: via `pages` rows that are direct `TASK_LIST` children of the list's page (`pages.parentId = pageId AND pages.type = 'TASK_LIST' AND pages.isTrashed = false`). The old per-row loop trusted every submitted task id blindly with no scoping at all.

I scoped the new write to `inArray(taskItems.pageId, childPageIds)` — reusing that exact, already-established derivation rather than guessing at a new one. `lockedBatchReorder`'s returned locked-ids are compared against the submitted plan's ids; a mismatch (an id outside the list's scope) now returns `400 { error: 'Invalid task IDs' }` instead of the previous code's implicit any-id-goes behavior. Since the original route had zero scope validation, this is a net-positive tightening, not a behavior regression.

## Test plan

- [x] New test: single tasks-list reorder issues exactly one batched `lockedBatchReorder` call regardless of task count (was previously N sequential `tx.update` calls) — proves the batching fix.
- [x] New test: a submitted task id outside the derived scope returns `400 Invalid task IDs` and does not broadcast.
- [x] Updated existing test: empty `tasks: []` array opens no transaction (explicit no-op, matches the epic's spec for this phase).
- [x] All pre-existing assertions in `route.test.ts` still pass (auth, permission, validation, broadcast, activity logging).
- [x] `bun run typecheck` — 16/16 tasks green.
- [x] `bun run test` (apps/web) — 990/994 files pass; 3 pre-existing, unrelated failures (missing local test-DB role `"test"` in this sandbox + one flaky date-boundary test in `grouping.test.ts`), none touching the reorder route.

PageSpace epic page: `j44e35jwzlhr54fbmruk3k4i` ("Task Board Query & Reorder Safety", Phase 4 of 8).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01AnsHCW5mvGrTi66fBm3HUU

### CODE REVIEW

File: apps/web/src/app/api/pages/[pageId]/tasks/reorder/route.ts [hunk 1]
```diff
@@ -64,14 +65,48 @@ export async function PATCH(
...
+    // this list's page. Scoping the write to that set closes a gap the old
+    // per-row loop never checked (it trusted every submitted id blindly).
+    const childPages = await db
+      .select({ id: pages.id })
+      .from(pages)
+      .where(and(
+        eq(pages.parentId, pageId),
+        eq(pages.type, 'TASK_LIST'),
+        eq(pages.isTrashed, false),
+      ));
```
[the remark this answers is not in the document]
Comment (2witstudios) [reply apps/web/src/app/api/pages/[pageId]/tasks/reorder/route.ts:85]: Fixed in fast-follow #2163 — passes the `pages` query builder directly as `inArray`'s second argument (matching the existing `fetchEnrichedTasks` pattern in `task-helpers.ts`) instead of awaiting/materializing the child page ids into an array first. This PR had already merged by the time this review landed, so the fix is a separate PR rather than a push to this (closed) branch.
